core/chain_makers.go

```diff
@@ -103,10 +104,17 @@ func (b *BlockGen) AddTxWithChain(bc *BlockChain, tx *types.Transaction) {
 		b.SetCoinbase(common.Address{})
 	}
 	b.statedb.Prepare(tx.Hash(), len(b.txs))
-	receipt, err := ApplyTransaction(b.config, bc, &b.header.Coinbase, b.gasPool, b.statedb, b.header, tx, &b.header.GasUsed, vm.Config{})
+	receipt, accesses, err := ApplyTransaction(b.config, bc, &b.header.Coinbase, b.gasPool, b.statedb, b.header, tx, &b.header.GasUsed, vm.Config{})
 	if err != nil {
 		panic(err)
 	}
+	if accesses != nil {
+		if b.statedb.Witness() != nil {
+			b.statedb.Witness().Merge(accesses)
+		} else {
+			b.statedb.SetWitness(accesses)
+		}
+	}
```

Comment:
simplification here: since the statedb now contains an `AccessWitness`, one can remove it from the signature of `ApplyTransaction` and directly merge it there.